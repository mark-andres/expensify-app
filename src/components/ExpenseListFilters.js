import React from 'react'
import { connect } from 'react-redux'
import { DateRangePicker } from 'react-dates'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }
    onSortChange = (e) => {
        switch (e.target.value) {
            case 'date':
                this.props.sortByDate()
                break
            case 'amount':
                this.props.sortByAmount()
                break
        }
    }
    render() {
        return (
            <div>
                <input 
                    type="text" 
                    value={this.props.filters.text} 
                    onChange={this.onTextChange}
                />
                <select 
                    value={this.props.filters.sortBy}
                    onChange={this.onSortChange}
                >
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
                    endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
                    onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
                    focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
       )
    }
}

const mapDispatchToProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (textFilter) => dispatch(setTextFilter(textFilter)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount())
})

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)