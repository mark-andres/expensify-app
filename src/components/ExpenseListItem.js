import React from 'react'
import { NavLink } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

const ExpenseListItem = ({ id, description, amount, createdAt, note }) => (
    <div>
        <h3><NavLink to={`/edit/${id}`}>{description}</NavLink></h3>
        <p>
            {numeral(amount / 100).format('$0,0.00')} 
            - 
            {note} 
            - 
            {moment(createdAt).format("dddd, MMMM Do YYYY")}}
        </p>
    </div> 
)

export default ExpenseListItem