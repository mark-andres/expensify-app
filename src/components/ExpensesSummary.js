import React from 'react'
import { connect } from 'react-redux'
import selectExpensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

export const ExpensesSummary = (props) => (
    <p>
        Viewing {props.count} expenses totalling {numeral(props.total / 100).format('$0,0.00')}
    </p>
)

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters)
    return {
        count: expenses.length,
        total: selectExpensesTotal(expenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)