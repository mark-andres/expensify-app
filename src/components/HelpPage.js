import React from 'react'

const HelpPage = () => (
    <div>
        This is from my help componet
    </div>
)

export default HelpPage