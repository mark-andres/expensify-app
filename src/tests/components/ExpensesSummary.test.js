import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'
import selectExpensesTotal from '../../selectors/expenses-total'

test('should render ExpensesSummary correctly with 0 expenses', () => {
    const wrapper = shallow(<ExpensesSummary count={0} total={0} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary correctly with expenses', () => {
    const wrapper = shallow(<ExpensesSummary count={expenses.length} total={selectExpensesTotal(expenses)} />)
    expect(wrapper).toMatchSnapshot()
})