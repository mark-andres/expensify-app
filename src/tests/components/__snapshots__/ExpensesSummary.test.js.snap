// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with 0 expenses 1`] = `
<p>
  Viewing 
  0
   expenses totalling 
  $0.00
</p>
`;

exports[`should render ExpensesSummary correctly with expenses 1`] = `
<p>
  Viewing 
  3
   expenses totalling 
  $1,141.95
</p>
`;
